aloe/result.py

```diff
@@ -0,0 +1,216 @@
+# Aloe - Cucumber runner for Python based on Lettuce and Nose
+# Copyright (C) <2015> Danielle Madeley <danielle@madeley.id.au>
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""
+Cucumber-esque outputter for Nose.
+"""
+
+from __future__ import print_function
+from __future__ import unicode_literals
+from __future__ import division
+from __future__ import absolute_import
+# pylint:disable=redefined-builtin, unused-wildcard-import, wildcard-import
+from builtins import *
+# pylint:enable=redefined-builtin, unused-wildcard-import, wildcard-import
+from future import standard_library
+standard_library.install_aliases()
+
+from contextlib import contextmanager
+from functools import wraps
+
+from aloe.registry import (
+    CallbackDecorator,
+    CALLBACK_REGISTRY,
+    PriorityClass,
+)
+from aloe.strings import represent_table
+from aloe.tools import hook_not_reentrant
+from blessings import Terminal
+from nose.result import TextTestResult
+
+# A decorator to add callbacks which wrap the steps looser than all the other
+# callbacks.
+# pylint:disable=invalid-name
+outer_around = CallbackDecorator(CALLBACK_REGISTRY, 'around',
+                                 priority_class=PriorityClass.DISPLAY)
+# pylint:enable=invalid-name
+
+
+TERMINAL = [None]  # global reference to the OutputTerminal
+
+
+class OutputTerminal(Terminal):
```

Comment:
Don't like the name (are there any other kinds of terminals? can we say `class Terminal(blessings.Terminal)`?) but not fussed.